File: web/models/connection.php
<?php

namespace models;

use Medoo\Medoo;
use Exception;
use PDO;

// Coneccion con la bd con patron de diseño Singleton
abstract class Connection {
    private static $database = null;

    public static function getDatabase() {
        if (self::$database == null) {
            try {
                // Dentro de una carpeta 'web' crear una nueva carpeta llamada 'config' y ahi adentro un archivo llamado db-config.conf
                // Dentro de este escribir toda la informacion de configuracion de la db, separada solo por punto y coma (;)
                // De la siguiente forma:
                //Ej: localhost;laboratorio;root;;3307; 
                $config = explode(';', file_get_contents('config/db-config'));
                $database = new Medoo([
                    'type' => 'mysql',
                    'host' => $config[0],
                    'database' => $config[1],
                    'username' => $config[2],
                    'password' => $config[3],
                    'port' => $config[4],

                    'error' => PDO::ERRMODE_SILENT,
                ]);
            } catch (Exception $e) {
                throw new Exception($e);
            }
        }
        return $database;
    }
}

File: python/generador_numeros/models/Connection.py
import sys
import mariadb
from . import configdb
#Crear archivo configdb.py con las siguientes constantes(cada una con los valores correspondientes a tu base de datos):
#USER='root'
#PASS=''
#DATABASE='laboratorio'
#HOST='localhost'
#PORT=3307

#Coneccion con la bd con patron de diseño Singleton
class Connection:
    database = None
    
    @staticmethod
    def getConnection():
        if(Connection.database == None):
            try:
                Connection.database = mariadb.connect(
                    user=configdb.USER,
                    password=configdb.PASS,
                    host=configdb.HOST,
                    port=configdb.PORT,
                    database=configdb.DATABASE
                )
            except mariadb.Error as e:
                print(f"Error al conectar con la base de datos: {e}")
                sys.exit(1)
                
        return Connection.database.cursor()
    